developing/structure/transaction.md

```diff
@@ -0,0 +1,1152 @@
+              },
+              {
+                "key": "sender",
+                "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+              }
+            ]
+          },
+```
+
+##### Fourth Event
+
+The fourth event is `tokens_swapped`. It summarizes the actual swaps that took place. Each set of 4 consecutive
+attribute pairs goes together:
+
+```json
+          {
+            "type": "token_swapped",
+            "attributes": [
+              {
+                "key": "module",
+                "value": "gamm"
+              },
+              {
+                "key": "sender",
+                "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+              },
+              {
+                "key": "pool_id",
+                "value": "604"
+              },
+              {
+                "key": "tokens_in",
+                "value": "15000000uosmo"
+              },
+              {
+                "key": "tokens_out",
+                "value": "289236271ibc/987C17B11ABC2B20019178ACE62929FE9840202CE79498E29FE8E5CB02B7C0A4"
+              },
+              {
+                "key": "module",
+                "value": "gamm"
+              },
+              {
+                "key": "sender",
+                "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+              },
+              {
+                "key": "pool_id",
+                "value": "611"
+              },
+              {
+                "key": "tokens_in",
+                "value": "289236271ibc/987C17B11ABC2B20019178ACE62929FE9840202CE79498E29FE8E5CB02B7C0A4"
+              },
+              {
+                "key": "tokens_out",
+                "value": "3802225ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2"
+              },
+              {
+                "key": "module",
+                "value": "gamm"
+              },
+              {
+                "key": "sender",
+                "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+              },
+              {
+                "key": "pool_id",
+                "value": "1"
+              },
+              {
+                "key": "tokens_in",
+                "value": "3802225ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2"
+              },
+              {
+                "key": "tokens_out",
+                "value": "15003605uosmo"
+              }
+            ]
+          },
+```
+
+The 3 groupings are:
+
+1. The module `gamm` executed `15000000uosmo` from `osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89` swapped into `pool_id 604` in exchange for `289236271ibc/987C17B11ABC2B20019178ACE62929FE9840202CE79498E29FE8E5CB02B7C0A4`
+1. The module `gamm` executed `289236271ibc/987C17B11ABC2B20019178ACE62929FE9840202CE79498E29FE8E5CB02B7C0A4` from `osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89` swapped into `pool_id 611` in exchange for `3802225ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2`
```

Comment:
just change from 1 1 1 to 1 2 3

Reply:
Markdown does it automatically when rendering. I know people will find it funny, but this may be my single favoruite markdown feature.

Reply:
Vuepress sadly does not automatically render it this way, when running your PR locally it showed as 111 still so I changed it after I merged